test/integration/requiredOptional.spec.ts

```diff
@@ -0,0 +1,160 @@
+      const result = await client.implicit.putOptionalBody();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle getOptionalGlobalQuery", async () => {
+      const result = await client.implicit.getOptionalGlobalQuery();
+      assert.equal(result._response.status, 200);
+    });
+  });
+
+  describe("Explicit Optional", () => {
+    it("should handle postOptionalArrayHeader", async () => {
+      const result = await client.explicit.postOptionalArrayHeader();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalArrayParameter", async () => {
+      const result = await client.explicit.postOptionalArrayParameter();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalArrayProperty", async () => {
+      const result = await client.explicit.postOptionalArrayProperty();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalClassParameter", async () => {
+      const result = await client.explicit.postOptionalClassParameter();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalClassProperty", async () => {
+      const result = await client.explicit.postOptionalClassProperty();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalIntegerHeader", async () => {
+      const result = await client.explicit.postOptionalIntegerHeader();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalIntegerParameter", async () => {
+      const result = await client.explicit.postOptionalIntegerParameter();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalIntegerProperty", async () => {
+      const result = await client.explicit.postOptionalIntegerProperty();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalStringHeader", async () => {
+      const result = await client.explicit.postOptionalStringHeader();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalStringParameter", async () => {
+      const result = await client.explicit.postOptionalStringParameter();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postOptionalStringProperty", async () => {
+      const result = await client.explicit.postOptionalStringProperty();
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should handle postRequiredArrayHeader", async () => {
+      try {
+        await client.explicit.postRequiredArrayHeader(null as any);
+        assert.fail("Expected client to throw");
+      } catch (error) {
+        assert(
+          error.message.indexOf("cannot be null or undefined") !== -1,
+          "Expected error to contain 'cannot be null or undefined'"
+        );
```

Comment:
Perhaps the following is more idiomatic:
```suggestion
       assert.include(error.message, "cannot be null or undefined");
```
If you agree, please apply it to the rest.

Reply:
I had to revert back as it looks like `assert.include` only works for arrays and not for strings 😞

Reply:
weird, it works in my validation [PR](https://github.com/Azure/autorest.typescript/pull/740/files#r503975225).